apps/server/src/modules/rocketchat-user/service/rocket-chat-user.service.ts

```diff
@@ -53,63 +53,96 @@ export class RocketChatUserService implements DeletionService, IEventHandler<Use
 		);
 		const rocketChatUser = await this.rocketChatUserRepo.findByUserId(userId);
 
-		if (rocketChatUser.length > 0) {
-			try {
-				const [, rocketChatUserDeleted] = await Promise.all([
-					this.rocketChatService.deleteUser(rocketChatUser[0].username),
-					this.rocketChatUserRepo.deleteByUserId(rocketChatUser[0].userId),
-				]);
+		if (rocketChatUser === null) {
+			const result = this.buildResultAndLog(0, 0, userId, 'RocketChat user already deleted', StatusModel.FINISHED);
 
-				const result = DomainDeletionReportBuilder.build(
-					DomainName.ROCKETCHATUSER,
-					[DomainOperationReportBuilder.build(OperationType.DELETE, rocketChatUserDeleted, [rocketChatUser[0].id])],
-					[
-						DomainDeletionReportBuilder.build(DomainName.ROCKETCHATSERVICE, [
-							DomainOperationReportBuilder.build(OperationType.DELETE, 1, [rocketChatUser[0].username]),
-						]),
-					]
-				);
+			return result;
+		}
+
+		try {
+			await this.rocketChatService.deleteUser(rocketChatUser.username);
```

Comment:
the return value is not checked, nor the http status or anything or am i missing that?

Reply:
.

Reply:
@WojciechGrancow added the check for the response, but in our manual tests it turned out that any other case (any other than success) client throws an exception (and doesn't return a real response we've been expecting with `success` set to `false`) so the `try` contains the only possible, successful case. But ok, just to be 100% sure (if the client's behaviour will change), @WojciechGrancow will add this check once again 😊 . Will be pretty fake case (only achievable through mocks for now), but will secure us for any changes on the _Rocket.Chat_ side.

BTW I think we should consider refactoring the whole Rocket.Chat client - it's currently "embedded" into the service which is not compliant with the current architecture (it should be a separate client, probably located in the `infra` module). It also doesn't allow for verifying the HTTP status code which we could also want I think. But I think it should be done in a separate ticket as there are many other methods implemented in it and would require some regression tests to be sure we don't break anything.

Reply:
no need then unless it becomes clearer, wasn't obvious to me.